lms/djangoapps/certificates/badge_handler.py

```diff
@@ -0,0 +1,134 @@
+"""
+BadgeHandler object-- used to award Badges to users who have completed courses.
+"""
+import hashlib
+import logging
+import requests
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.core.urlresolvers import reverse
+from lazy.lazy import lazy
```

Comment:
Just as an observation, I usually did `from lazy import lazy`.  It turns out this is completely equivalent, but being lazy, I'll stick with the shorter form. :)